service/src/main/java/bio/terra/workspace/service/workspace/flight/delete/workspace/WorkspaceDeleteFlight.java

```diff
@@ -77,12 +77,17 @@ public WorkspaceDeleteFlight(FlightMap inputParameters, Object applicationContex
     addStep(
         new EnsureNoWorkspaceChildrenStep(appContext.getSamService(), userRequest, workspaceUuid));
 
-    addStep(
-        new DeleteAzureContextStep(appContext.getAzureCloudContextService(), workspaceUuid),
-        cloudRetryRule);
-    addStep(
-        new DeleteAwsContextStep(appContext.getAwsCloudContextService(), workspaceUuid),
-        cloudRetryRule);
+    // This step just does the DeleteCloudFinishStep.
+    // I do not think need this at all, since we are deleting the workspace object.
+    // but I am also thinking we should run these context deletes as separate flights.
```

Comment:
so what's the final plan? :)

Reply:
for the first option, if we don't delete workspace row and leave it broken, then cloudcontext will stuck in deleting i think ..

Reply:
Sorry, that is a note to myself. The follow-on PR will change the implementation of workspace delete and I plan to run the cloud context delete as a separate flight. I'll make that a proper TODO.